Xamarin.Forms.Platform.Android/AppCompat/MasterDetailContainer.cs

```diff
@@ -113,13 +113,17 @@ protected override void AddChildView(VisualElement childView)
 
 				new Handler(Looper.MainLooper).PostAtFrontOfQueue(() =>
 				{
+					// NOTE: If we delay ExecutePendingTransactions in any way in order to wait for the _pageContainer to be created,
+					// then we regress 44596/1373 by introducing the white screen flashing between main page changes to an MDP.
+					// However, we may need to restore this null check if we see crashes happening again.
+#if MASTERDETAILPAGECRASHES
 					if (_pageContainer == null)
 					{
 						// The view we're hosting in the fragment was never created (possibly we're already 
 						// navigating to another page?) so there's nothing to commit
 						return;
 					}
-
+#endif
 					FragmentManager.ExecutePendingTransactions();
```

Comment:
I tested this one using this scenario from #3187 

https://github.com/xamarin/Xamarin.Forms/blob/b83eba088229cee833e4abf6645745be1a12b84b/Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue2338.cs#L31 

Which causes a crash on this line of code if the if statement is missing.  So I'm pretty sure as is this PR will re-introduce the exception on this line of code.

I tested the flashing with #3187  and with this code and it doesn't appear to flash and/or crash on this line.

I realize this is going to master and the other to 3.1.0 so eventually it'll get merged up but I'm wondering if we should just take these changes and replay them on #3187 ?

Or I can just squash merge 3187 in right now since it's ready and then retarget this for 3.1.0?

Reply:
I don't want this going into 3.1.0, so let's just not merge this until 3187 makes its way to master.